lib/jsonapi/serializer.ex

```diff
@@ -188,7 +191,12 @@ defmodule JSONAPI.Serializer do
          {rel_view, rel_data, rel_url, conn},
          false = _remove_links
        ) do
-    Map.merge(encoded_data, %{links: %{self: rel_url, related: rel_view.url_for(rel_data, conn)}})
+    if rel_data do
+      related_url = rel_view.url_for(rel_data, conn)
+      Map.merge(encoded_data, %{links: %{self: rel_url, related: related_url}})
+    else
+      Map.merge(encoded_data, %{links: %{self: rel_url}})
```

Comment:
We might be able to generate the `related` URL if there's a foreign key in the data